src/types.ts

```diff
@@ -166,9 +166,15 @@ export type AppState = {
     | "backgroundColorPicker"
     | "strokeColorPicker"
     | null;
-  openSidebar: "library" | "customSidebar" | null;
+  openSidebar: { name: string; tab?: string } | null;
   openDialog: "imageExport" | "help" | "jsonExport" | null;
-  isSidebarDocked: boolean;
+  /**
+   * Reflects user preference for whether the default sidebar should be docked.
+   *
+   * NOTE this is only a user preference and does not reflect the actual docked
+   * state of the sidebar, because the host apps can override this through
+   * a Sidebar prop, which is not reflected back to the appState. */
+  defaultSidebarDockedPreference: boolean;
```

Comment:
I don't see `isSidebarDocked` documented in our docs so is this for internal use ?
Is `defaultSidebarDockedPreference` only related to the default sidebar ?

Reply:
We don't really document individual appState properties, no?

> Is `defaultSidebarDockedPreference` only related to the default sidebar ?

Yes

Reply:
> We don't but if its used in the component API then we should document unless its some hidden property which we don't recommend to use.

Nothing against that, but the `isSidebarDocked` wasn't introduced in this PR, and the new docs aren't written yet.

> Moreover why. a component shouldn't depend on Excalidraw app State ? This sort of is confusing or may be I am missing something
can we not make it controlled in that case so it's completely controlled by host via a prop /callback?
I think you have mentioned something in regards to this but I am still not clear so would be great if you could elaborate.

They can depend on Excalidraw state. This case is different that the default Sidebar is both controlled via appState and the `docked` prop. So as per the notes above, we can either make it (1) semi-controlled or (2) fully controlled.

The (1) didn't pan out because we wouldn't be able to properly update appState based on `props.docked` (and it also seems like a bad pattern).

So I went with (2), renaming the appState to make it clear it's just a user preference, and not necessarily reflect the actual state.